core/src/test/scala/org/broadinstitute/dsde/rawls/monitor/workspace/runners/BPMBillingProjectDeleteRunnerSpec.scala

```diff
@@ -0,0 +1,464 @@
+                                            Timestamp.from(Instant.now())
+      )
+
+    whenReady(runner(monitorRecord))(_ shouldBe WorkspaceManagerResourceMonitorRecord.Incomplete)
+    verify(billingRepository).updateCreationStatus(
+      ArgumentMatchers.eq(billingProjectName),
+      ArgumentMatchers.eq(DeletionFailed),
+      ArgumentMatchers.any[Some[String]]()
+    )
+  }
+
+  it should "set an error status and message on the project and return incomplete when landing zone call returns 500" in {
+    val ctx = mock[RawlsRequestContext]
+    val wsmDao = mock[WorkspaceManagerDAO]
+    val wsmExceptionMessage = "looking for this to be reported in the billing project message"
+    val landingZoneId = "7db89d7c-ef8b-4eaa-aef8-a62bd66cb095"
+    val monitorRecord: WorkspaceManagerResourceMonitorRecord = WorkspaceManagerResourceMonitorRecord(
+      UUID.randomUUID(),
+      JobType.AzureBillingProjectDelete,
+      None,
+      Some(billingProjectName.value),
+      Some(userEmail),
+      Timestamp.from(Instant.now())
+    )
+
+    when(
+      wsmDao.getDeleteLandingZoneResult(
+        ArgumentMatchers.eq(monitorRecord.jobControlId.toString),
+        ArgumentMatchers.eq(UUID.fromString(landingZoneId)),
+        ArgumentMatchers.any()
+      )
+    )
+      .thenAnswer(_ => throw new bio.terra.workspace.client.ApiException(500, wsmExceptionMessage))
+    val billingRepository = mock[BillingRepository]
+    when(billingRepository.getLandingZoneId(ArgumentMatchers.eq(billingProjectName))(any()))
+      .thenReturn(Future.successful(Some(landingZoneId)))
+    when(
+      billingRepository.updateCreationStatus(
+        ArgumentMatchers.eq(billingProjectName),
+        ArgumentMatchers.eq(DeletionFailed),
+        ArgumentMatchers.any[Some[String]]()
+      )
+    ).thenAnswer { invocation =>
+      val message: Option[String] = invocation.getArgument(2)
+      assert(message.get.contains(wsmExceptionMessage))
+      Future.successful(1)
+    }
+    val runner = spy(
+      new BPMBillingProjectDeleteRunner(
+        mock[SamDAO],
+        mock[GoogleServicesDAO],
+        wsmDao,
+        billingRepository,
+        mock[BillingProjectLifecycle]
+      )
+    )
+    doReturn(Future.successful(ctx)).when(runner).getUserCtx(ArgumentMatchers.eq(userEmail))(ArgumentMatchers.any())
+
+    whenReady(runner(monitorRecord))(_ shouldBe WorkspaceManagerResourceMonitorRecord.Incomplete)
+
+    verify(billingRepository).updateCreationStatus(
+      ArgumentMatchers.eq(billingProjectName),
+      ArgumentMatchers.eq(DeletionFailed),
+      ArgumentMatchers.any[Some[String]]()
+    )
+  }
+
+  it should "set an error status and message from the error report for failed jobs" in {
+    val ctx = mock[RawlsRequestContext]
+    val wsmDao = mock[WorkspaceManagerDAO]
+    val wsmExceptionMessage = "looking for this to be reported in the billing project message"
+    val landingZoneId = "7db89d7c-ef8b-4eaa-aef8-a62bd66cb095"
```

Comment:
If the landing zone ID is not meaningful, I think it's better to create a random one.

Reply:
I've been reading up on pact, and how random values that change across test runs should be avoided. 
It isn't applicable here, and this project is a long way from being able to effectively use contracts, but it stuck in my head. 
I don't have a problem generating a random uuid here, if it makes it clearer.